### PR TITLE
storybook@v0.9.0 – Updating to latest storybook version and adding typography docs #trivial

### PR DESCRIPTION
**Pull request by ashleynolan**
### Added
- Config for parsing `.mdx` files which can be used for markdown documentation with components.

### Changed
- Updated to Storybook `v6` (and updated stories across other packages as part of the migration).

### CODE REVIEW

**Review by minrwhite — DISMISSED** [review]

File: packages/f-footer/stories/footer.stories.js [hunk 1]
```diff
@@ -31,6 +31,4 @@ export const FooterComponent = () => ({
     template: '<vue-footer :showCourierLinks="showCourierLinks" :locale="locale" />'
 });
 
-FooterComponent.story = {
-    name: 'f-footer'
-};
+FooterComponent.storyName = 'f-footer';
```
Comment (xander-marjoram) [inline packages/f-footer/stories/footer.stories.js:34]: Nit: It does irk me that some of these have `Component` and some have `component`

File: packages/storybook/package.json [hunk 2]
```diff
@@ -9,15 +9,15 @@
...
   "devDependencies": {
     "@percy/storybook": "3.3.0",
-    "@storybook/addon-a11y": "^5.3.19",
-    "@storybook/addon-actions": "5.3.19",
-    "@storybook/addon-knobs": "5.3.19",
-    "@storybook/addon-links": "5.3.19",
-    "@storybook/addon-notes": "5.3.19",
-    "@storybook/vue": "5.3.19",
+    "@storybook/addon-a11y": "6.0.5",
+    "@storybook/addon-actions": "6.0.5",
```
Comment (xander-marjoram) [inline packages/storybook/package.json]: Do we still need this as a dependency if we're not calling it in main.js?

Comment (ashleynolan) [reply packages/storybook/package.json]: Nope - it's now part of the essentials package, so I'll delete

File: packages/f-footer/stories/footer.stories.js (hunk 1 above)
Comment (ashleynolan) [reply packages/f-footer/stories/footer.stories.js:34]: Git blame whoever named the component 😉  Guessing it's just been copy-pasted from the first content card that was created?

// cc @benwhite-deltas @asherstoppard  One for your next PR with any content card updates? 😆

File: packages/storybook/package.json (hunk 2 above)
Comment (ashleynolan) [reply packages/storybook/package.json]: Have just updated

File: packages/storybook/config/storybook/main.js [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
-    '@storybook/addon-knobs',
-    '@storybook/addon-links',
-    '@storybook/addon-notes',
-    '@storybook/addon-a11y/register'
-  ]
-};
\ No newline at end of file
+    stories: [
+        '../../../**/*.stories.@(js|mdx)',
+        '../../../../stories/*.stories.@(js|mdx)'
```
Comment (DamianMullins) [inline packages/storybook/config/storybook/main.js:4]: Does the previous glob not pick this path up?

**Review by DamianMullins — APPROVED** [review]

**Comment by DamianMullins** [comment]
@ashleynolan I assume the Percy checks are OK as it looks like a padding adjustment has shifted them all slightly?

File: packages/storybook/config/storybook/main.js (hunk 3 above)
Comment (ashleynolan) [reply packages/storybook/config/storybook/main.js:4]: Weirdly, I don't think it did when I was playing with the patterns.

This was the recommended pattern from the docs as apparently the old version of storybook didn't implement `proper globs` 🤷‍♂️ .  Apparently `@` matches exactly one of the patterns in the brackets according to the spec. I'd never seen it used before though…!

**Comment by ashleynolan** [comment]
@DamianMullins Yeh, I think this must be a storybook v6 padding addition

**Comment by ashleynolan** [comment]
@benwhite-deltas @xander-marjoram Any chance you could give this a second look?

**Review by minrwhite — APPROVED** [review]

**Pull request merged**